js/ftx.js

```diff
@@ -2546,6 +2562,22 @@ module.exports = class ftx extends Exchange {
         if (borrowRateHistory === undefined) {
             throw new BadRequest (this.id + '.fetchBorrowRateHistory returned no data for ' + code);
         } else {
+            const resultLength = borrowRateHistory.length;
+            if (resultLength === limit - 1) {
+                const hourLength = 3600000;
+                const recentHour = parseInt (this.milliseconds () / hourLength) * hourLength;
+                const penultimateHour = recentHour - hourLength;
+                const lastTimestamp = borrowRateHistory[0]['timestamp'];
+                if (lastTimestamp === penultimateHour) {
+                    borrowRateHistory.push ({
+                        'currency': code,
+                        'rate': undefined,
+                        'timestamp': recentHour,
+                        'datetime': this.iso8601 (recentHour),
+                        'info': {},
+                    });
+                }
+            }
```

Comment:
Let me know what you think of this part, it's here because sometimes the exchange doesn't return the rate for the most recent hour if the you call the method a few minutes after the hour. Because the api params only include `start_time` and `end_time`, and the way that fetchBorrowRateHistories calculates these parameters based on the limit, if the most recent rate isn't returned, it means the response will be 1 less than the limit, I thought this could be confusing and might make people think theres a bug, so if the length of the response is equal to the limit -1, and the second most recent rate is there but not the most recent, then I add a record for the most recent rate, but set the rate to `undefined`

---------------------

example.

if you call this method a 8:02 sometimes you'll get a response like below

```
% ftx fetchBorrowRateHistory USDT undefined 5

ftx.fetchBorrowRateHistory (USDT, , 5)

currency |        rate |     timestamp |                  datetime
------------------------------------------------------------------
    USDT | 0.000003078 | 1649077200000 | 2022-04-04T13:00:00+00:00
    USDT | 0.000003078 | 1649073600000 | 2022-04-04T12:00:00+00:00
    USDT | 0.000003078 | 1649070000000 | 2022-04-04T11:00:00+00:00
    USDT | 0.000003078 | 1649066400000 | 2022-04-04T10:00:00+00:00
4 objects
```
Which has a length 1 less than the limit. 

The 16 lines above make the response
```
% ftx fetchBorrowRateHistory USDT undefined 5

ftx.fetchBorrowRateHistory (USDT, , 5)

currency |        rate |     timestamp |                  datetime
------------------------------------------------------------------
    USDT | undefined   | 1649080800000 | 2022-04-04T14:00:00+00:00
    USDT | 0.000003078 | 1649077200000 | 2022-04-04T13:00:00+00:00
    USDT | 0.000003078 | 1649073600000 | 2022-04-04T12:00:00+00:00
    USDT | 0.000003078 | 1649070000000 | 2022-04-04T11:00:00+00:00
    USDT | 0.000003078 | 1649066400000 | 2022-04-04T10:00:00+00:00
5 objects
```

Reply:
I don't think we should alter the data from the exchange. We don't do that for OHLCVs, therefore we should not do that for borrow rate histories either.